pkg/controller/migmigration/backup.go

```diff
@@ -139,12 +160,86 @@ func (t Task) getBackup(labels map[string]string) (*velero.Backup, error) {
 }
 
 // Get whether a backup has completed on the source cluster.
-func (t Task) hasBackupCompleted(backup *velero.Backup) (bool, []string) {
+func (t *Task) hasBackupCompleted(backup *velero.Backup) (bool, []string) {
 	completed := false
 	reasons := []string{}
+	progress := []string{}
+
+	pvbs := t.getPodVolumeBackupsForBackup(backup)
+
+	getPodVolumeBackupsProgress := func(pvbList *velero.PodVolumeBackupList) (progress []string) {
+		if pvbList == nil {
+			return
+		}
+		for _, pvb := range pvbList.Items {
+			switch pvb.Status.Phase {
+			case velero.PodVolumeBackupPhaseInProgress:
+				progress = append(
+					progress,
+					fmt.Sprintf(
+						"PodVolumeBackup %s/%s: %s out of %s backed up",
+						pvb.Namespace,
+						pvb.Name,
+						bytesToSI(pvb.Status.Progress.BytesDone),
+						bytesToSI(pvb.Status.Progress.TotalBytes)))
+			case velero.PodVolumeBackupPhaseCompleted:
+				progress = append(
+					progress,
+					fmt.Sprintf(
+						"PodVolumeBackup %s/%s: Completed (%s backed up)",
+						pvb.Namespace,
+						pvb.Name,
+						bytesToSI(pvb.Status.Progress.TotalBytes)))
+			case velero.PodVolumeBackupPhaseFailed:
+				progress = append(
+					progress,
+					fmt.Sprintf(
+						"PodVolumeBackup %s/%s: Failed",
+						pvb.Namespace,
+						pvb.Name))
+			default:
+				progress = append(
+					progress,
+					fmt.Sprintf(
+						"PodVolumeBackup %s/%s: Waiting for ongoing volume backup(s) to complete",
+						pvb.Namespace,
+						pvb.Name))
+			}
+		}
```

Comment:
I would actually consider sorting the `progress` array before returning in order to reduce flapping due to the different order of `pvbList.Items` returned

Reply:
@alaypatel07 I have not seen that behaviour. The order remains same, maybe that's a wrong assumption to make. Unsure. Sorting based on ongoing phase would be best I think (InProgress at the top, followed by Completed, Failed and Waiting)

Reply:
@pranavgaikwad yes I think it will take a lot of time to observe that behaviour but I don't think the apiserver guarantees order. We should safeguard ourselves against that wherever we can

Reply:
not too opinionated of what would be the best place, will leave that decision to you :)

Reply:
+1 on sorting at the end of the loop. That way we can make sure that the order of PVB remains unchanged.

Reply:
+1 on sorting at the end of loop based on `pvb.Namespace/pvb.Name` to keep the order same everytime.